x-pack/plugins/uptime/server/lib/adapters/monitors/elasticsearch_monitors_adapter.ts

```diff
@@ -187,35 +180,42 @@ export class ElasticsearchMonitorsAdapter implements UMMonitorsAdapter {
       },
     };
 
-    const queryResult = await this.database.search(request, params);
-    const hostBuckets = get(queryResult, 'aggregations.urls.buckets', []);
-    const monitorStatuses = hostBuckets.map(bucket => {
-      const latest = get(bucket, 'latest.hits.hits', []);
-      return latest.reduce(
-        (acc, doc) => {
-          const status = get(doc, '_source.monitor.status', null);
-          if (statusFilter && statusFilter !== status) {
-            return acc;
-          }
-          if (status === 'up') {
-            acc.up += 1;
-          } else {
-            acc.down += 1;
-          }
-          return acc;
-        },
-        { up: 0, down: 0 }
-      );
-    });
-    const { up, down } = monitorStatuses.reduce(
-      (acc, status) => {
-        acc.up += status.up;
-        acc.down += status.down;
-        return acc;
-      },
-      // @ts-ignore TODO update typings and remove this comment
-      { up: 0, down: 0 }
-    );
+    let up: number = 0;
+    let down: number = 0;
+    let searchAfter: any = null;
+
+    do {
+      if (searchAfter) {
+        set(params, 'body.aggs.ids.composite.after', searchAfter);
+      }
+
+      const queryResult = await this.database.search(request, params);
+      const idBuckets = get(queryResult, 'aggregations.ids.buckets', []);
+
+      if (idBuckets.length === 0) {
+        break;
+      }
+
+      idBuckets.forEach(bucket => {
+        const latest = get(bucket, 'latest.hits.hits', []);
+        return latest.forEach(
+          doc => {
+            const status = get(doc, '_source.monitor.status', null);
+            if (statusFilter && statusFilter !== status) {
+              if (status === 'up') {
```

Comment:
I'm fine with this as I agree now with what you said, we'll use less memory if we just increment the count.